api/resources_portal/views/material_request_issue.py

```diff
@@ -0,0 +1,209 @@
+from django.forms.models import model_to_dict
+from rest_framework import serializers, viewsets
+from rest_framework.permissions import BasePermission, IsAuthenticated
+from rest_framework.response import Response
+
+from guardian.shortcuts import get_objects_for_user
+
+from resources_portal.models import (
+    MaterialRequest,
+    MaterialRequestIssue,
+    Notification,
+    Organization,
+)
+from resources_portal.views.relation_serializers import MaterialRequestRelationSerializer
+
+MATERIAL_ARCHIVED_ERROR = "Cannot open issues for requests whose materials have been archived."
+REQUEST_UNFULFILLED_ERROR = "Cannot open issues for requests which haven't been fulfilled."
+
+
+class MaterialRequestIssueSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = MaterialRequestIssue
+        fields = (
+            "id",
+            "description",
+            "status",
+            "material_request",
+            "created_at",
+            "updated_at",
+        )
+        read_only_fields = (
+            "id",
+            "created_at",
+            "updated_at",
+            "material_request",
+        )
+
+
+class MaterialRequestIssueDetailSerializer(MaterialRequestIssueSerializer):
+    material_request = MaterialRequestRelationSerializer()
+
+
+class IsRequester(BasePermission):
+    def has_permission(self, request, view):
+        material_request = MaterialRequest.objects.get(
+            pk=view.kwargs["parent_lookup_material_request"]
+        )
+
+        return request.user == material_request.requester
+
+
+class IsAdminUser(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return request.user.is_superuser
+
+
+class IsRequesterOrIsInOrg(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return (
+            request.user == obj.material_request.requester
+            or request.user in obj.material_request.material.organization.members.all()
+        )
+
+
+# def send_material_request_issue_notif(notif_type, issue, notified_user):
```

Comment:
Do these still need to be here? Are they linked to the TODO below?

Reply:
Nah I originally copied over `material_request.py` and wasn't sure if I was gonna want to use these here or not, but I did the notifications another way.